src/index.js

```diff
@@ -12,6 +13,33 @@ const RequestShortener = require("webpack/lib/RequestShortener");
 const ModuleFilenameHelpers = require("webpack/lib/ModuleFilenameHelpers");
 const uglify = require("uglify-js");
 
+const WARN_PATTERN = /\[.+:([0-9]+),([0-9]+)\]/;
+
+function ensureConditionFunc(condition, key) {
+	switch(typeof condition[key]) {
+		case "boolean": {
+			const b = condition[key];
+			condition[key] = () => b;
+			break;
+		}
+		case "function":
+			break;
+		case "string": {
+			if(condition[key] === "all") {
+				condition[key] = () => true;
+				break;
+			}
+			const regex = new RegExp(condition[key]);
+			condition[key] = (astNode, comment) => regex.test(comment.value);
+			break;
+		}
+		default: {
+			const defaultRegex = condition[key];
```

Comment:
That's how you write a switch case when you need a lexical declaration.

This has precedent in C/C++ as well.